src/porepy/models/run_models.py

```diff
@@ -49,36 +55,81 @@ def run_time_dependent_model(model, params: dict) -> None:
     if params.get("prepare_simulation", True):
         model.prepare_simulation()
 
+    # Change the position of the solver progress bar to 1, as the time progress bar
+    # will occupy position 0.
+    params.update({"progress_bar_position": 1})
+
     # Assign a solver
     solver: Union[pp.LinearSolver, pp.NewtonSolver]
+
     if model._is_nonlinear_problem():
         solver = pp.NewtonSolver(params)
     else:
         solver = pp.LinearSolver(params)
 
-    # Time loop
-    while model.time_manager.time < model.time_manager.time_final:
-        model.time_manager.increase_time()
-        model.time_manager.increase_time_index()
-        logger.info(
-            "\nTime step {} at time {:.1e} of {:.1e} with time step {:.1e}".format(
-                model.time_manager.time_index,
-                model.time_manager.time,
-                model.time_manager.time_final,
-                model.time_manager.dt,
+    # Progressbars turned on:
+    if params.get("progressbars", True):
+        # Redirect the root logger, s.t. no logger interferes with with the
+        # progressbars.
+        with logging_redirect_tqdm([logging.root]):
```

Comment:
Is it possible to use a decorator for this part, in a way that avoids code duplication (see various comments below)?

Reply:
I put it into a seperate function.